solidity/dashboard/src/services/token-grants.service.js

```diff
@@ -17,62 +22,88 @@ const fetchGrants = async (web3Context) => {
       yourAddress
     )
   )
+  const managedGrants = await fetchManagedGrants(web3Context)
   const grants = []
-
   for (const grantId of grantIds) {
-    const grantDetails = await contractService.makeCall(
-      web3Context,
-      TOKEN_GRANT_CONTRACT_NAME,
-      "getGrant",
-      grantId
-    )
-    if (!isSameEthAddress(yourAddress, grantDetails.grantee)) {
+    let grantDetails = {}
+    try {
+      grantDetails = await getGrantDetails(grantId, web3Context)
+    } catch (error) {
```

Comment:
if we don't use the `(error)`, maybe we can remove it?
Actually, not sure if we want to notify a user about failed grant details request?